Added the German localization support and updated the language selection `text` to `flag emoji`

Hedayat-Farahi786 opened the pull request:
**Hey @aykutkardas 👋**

**I've added the followings:**

* `German` localization support
* Changed the language selection `text` to `flag emojis`

Hope you like it!

<img src="https://i.imgur.com/UymYc0t.png" width="500"/>
<img src="https://i.imgur.com/Qxc1Bsk.png" width="500"/>

Emojis Credit: https://emojipedia.org

aykutkardas commented on src/localization/de/general.json:
```diff
@@ -0,0 +1,35 @@
+  "general.next": "Weiter",
+  "general.result": "Ergebnis",
+  "general.regex": "Regex",
+  "allgemein.text": "Text",
```
It's should be  `"general.text"`

Hedayat-Farahi786 replied:
Ohh, my bad 🤦‍♂️

Hedayat-Farahi786 replied:
I might got confused 😑

aykutkardas commented on src/components/LanguageSwitch/LanguageSwitch.module.css:
```diff
@@ -23,15 +23,17 @@
   top: 20px;
   z-index: 99;
-  right: 21px;
+  right: 0px;
```
![image](https://user-images.githubusercontent.com/7966133/153504582-461e81ce-6761-4d68-9c4b-f0f8f9e3f984.png)

It's should be `25px`


![image](https://user-images.githubusercontent.com/7966133/153504736-91207009-56b3-41b6-8e28-a8ea884eb375.png)

Hedayat-Farahi786 replied:
Yeah, that looks great! 🚀

Hedayat-Farahi786 replied:
Or we can just use `left: 0px` instead of `right` value!

aykutkardas commented on src/components/LanguageSwitch/LanguageSwitch.module.css:
```diff
@@ -63,9 +65,13 @@
   display: inline-block;
 }
 
+.LanguageSwitchListItem:nth-child(1) {
+  margin-top: 10px;
+}
```
This should be deleted. Instead, change the `top` value in the `.LanguageSwitchList` class to `30px`.

aykutkardas commented on src/localization/de/learn.json:
```diff
@@ -0,0 +1,163 @@
+  "steps.whatIsRegex.description": "Reguläre Ausdrücke sind eine Zeichenkette, die ein Suchmuster ausdrückt. Oft als `Regex` oder `Regexp` abgekürzt. Sie werden besonders verwendet, um Wörter in Texten zu finden oder zu ersetzen. Darüber hinaus können wir testen, ob ein Text den von uns festgelegten Regeln entspricht.\\n\\n Angenommen, Sie haben eine Liste mit Dateinamen und möchten nur Dateien m ...
+
+  "steps.basicMatchers.title": "Grundlegende Matcher",
+  "steps.basicMatchers.description": "Das zu findende Zeichen oder Wort wird direkt geschrieben. Es ähnelt einem normalen Suchvorgang. Um beispielsweise das Wort `neugierig` im Text zu finden, geben Sie dasselbe ein.",
```
`neugierig` This is the word intended to be in the text, so it should not be translated.

Expected: `curious`

aykutkardas commented on src/localization/de/learn.json:
```diff
@@ -0,0 +1,163 @@
+  "examples.rangeNumber.description": "Geben Sie einen Ausdruck ein, um Zahlen von `2` bis `7` im Text auszuwählen.\\n`2` und `7` sollten auch in diesem Bereich enthalten sein.",
+
+  "steps.repetitions.title": "Wiederholungen",
+  "steps.repetitions.description": "Einige Sonderzeichen werden verwendet, um anzugeben, wie oft ein Zeichen im Text wiederholt wird. Diese Sonderzeichen sind das Plus `+`, das Sternchen `*` und das Fragezeichen `? `.",
```
**Current:** \`? `

**Expected:** \`?`

aykutkardas commented on src/localization/de/learn.json:
```diff
@@ -0,0 +1,163 @@
+  "steps.nonCapturingGroupping.description": "Sie können einen Ausdruck gruppieren und sicherstellen, dass er nicht durch Referenzen erfasst wird. Unten sehen Sie beispielsweise zwei Gruppen. Die erste Gruppenreferenz, die wir mit `\\1` bezeichnen, zeigt jedoch tatsächlich die zweite an Gruppe, da die erste eine nicht einfangende Gruppe ist.",
+
+  "steps.pipeCharacter.title": "Pipe-Zeichen `|`",
+  "steps.pipeCharacter.description": "Damit kann festgelegt werden, dass ein Ausdruck in verschiedenen Ausdrücken enthalten sein kann. Daher werden alle möglichen Anweisungen durch das Pipe-Zeichen `|` getrennt geschrieben. Dies unterscheidet sich vom Zeichensatz `[abc]`, Zeichensätze arbeiten auf Zeichenebene. Alternativen befinden sich auf Ausdrucksebene. Beispielsweise würde der folgende Ausdr ...
```
This is the words intended to be in the text, so it should not be translated.
`katze`, `Katze`, `Ratte`

**Expected:** `cat`, `Cat`, `Rat`

aykutkardas commented on src/localization/de/learn.json:
```diff
@@ -0,0 +1,163 @@
+  "steps.withoutSpaceCharacter.description": "`\\S` wird verwendet, um Nicht-Leerzeichen zu finden.",
+
+  "steps.lookarounds.title": "Lookarounds",
+  "steps.lookarounds.description": "Wenn wir möchten, dass der Satz, den wir schreiben, vor oder nach einem anderen Satz steht, müssen wir \"umsehen\". Machen Sie den nächsten Schritt, um zu lernen, wie man \"umsieht\"." ,
```
Use `lookarounds` instead of `umsehen`

aykutkardas commented on src/localization/de/learn.json:
```diff
@@ -0,0 +1,163 @@
+  "steps.flagsCaseInsensitive.description": "Um die Groß-/Kleinschreibung des von uns geschriebenen Ausdrucks zu entfernen, müssen wir das Flag `Groß-/Kleinschreibung ignorieren` aktivieren.",
+  
+  "steps.greedyMatching.title": "Gieriger Abgleich",
+  "steps.greedyMatching.description": "Regex führt standardmäßig einen Greedy-Match durch. Das bedeutet, dass das Matchmaking so lang wie möglich sein wird. Sehen Sie sich das Beispiel unten an. Es bezieht sich auf alle Matches, die auf „r“ enden und beliebig sein können vorangestelltes Zeichen. Aber es hört nicht beim ersten Buchstaben `r` auf.",
```
I'm not sure if it's a correct word.

`Matchmaking`

Hedayat-Farahi786 replied:
`Gieriger` == `Greedy` 
`Heiratsvermittler` == `Matchmaking`

Hedayat-Farahi786 replied:
Abgleich is kinda like a common saying word!

aykutkardas commented:
> Yeah, your right! I made some typos even though I checked it all three times but still🤦‍♂️

I'm waiting for their fix. Then it's almost ready to merge. @Hedayat-Farahi786 ✨

Hedayat-Farahi786 commented:
**Hey @aykutkardas,**

- Fixed all the typos
- Added fade-in-out animation to the languages list
- Added the scale transformation to the Flags on Hover

![](https://i.imgur.com/JdAQScu.gif)
